x/cronos/keeper/msg_server.go

```diff
@@ -64,9 +64,12 @@ func (k msgServer) TransferTokens(goCtx context.Context, msg *types.MsgTransferT
 // UpdateTokenMapping implements the grpc method
 func (k msgServer) UpdateTokenMapping(goCtx context.Context, msg *types.MsgUpdateTokenMapping) (*types.MsgUpdateTokenMappingResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	admin := k.Keeper.GetParams(ctx).CronosAdmin
-	// if admin is empty, no sender could be equal to it
-	if admin != msg.Sender {
+	acc, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return nil, err
+	}
+	// check permission
+	if !k.HasPermission(ctx, acc, CanChangeTokenMapping) {
```

Comment:
maybe ok like this, but was thinking whether it'd make sense to move it to an ante handler, so that all permission-related checking is in one place (I think that SDK's generic circuit breaker will be an ante handler)?